src/rules/cognitive-complexity.ts

```diff
@@ -339,20 +338,47 @@ const rule: TSESLint.RuleModule<string, (number | 'metric' | 'sonar-runtime')[]>
     }
 
     function visitLogicalExpression(logicalExpression: TSESTree.LogicalExpression) {
+      if (isJsxShortCircuitNode(logicalExpression)) {
+        return;
+      }
+
       if (!consideredLogicalExpressions.has(logicalExpression)) {
         const flattenedLogicalExpressions = flattenLogicalExpression(logicalExpression);
 
         let previous: TSESTree.LogicalExpression | undefined;
         for (const current of flattenedLogicalExpressions) {
           if (!previous || previous.operator !== current.operator) {
-            const operatorTokenLoc = getFirstTokenAfter(logicalExpression.left, context)!.loc;
```

Comment:
why this change?

Reply:
That's the issue I mention in my comment above. All the locations raised by this function are the same instead of pointing to the flattened logical operators.

Reply:
FYI, I added a test case for this fix in the last commit.